### PR TITLE
Add error message for CollectionHandle init

### PR DESCRIPTION
**Pull request by SHeimlich**
Add error message for CollectionHandle init

### CODE REVIEW

File: java/arcs/core/entity/CollectionHandle.kt [hunk 1]
```diff
@@ -45,7 +45,9 @@ class CollectionHandle<T : Storable, R : Referencable>(
   private val storageAdapter = config.storageAdapter
 
   init {
-    check(spec.containerType == HandleContainerType.Collection)
+    check(spec.containerType == HandleContainerType.Collection) {
+      "HandleContainerType of ${name} does not match Spec containerType."
```
Comment (csilvestrini) [inline java/arcs/core/entity/CollectionHandle.kt:49]: You could be even clearer, and say something like: `"Collection container type required for CollectionHandle $name, but got ${spec.containerType}."`

**Review by csilvestrini — APPROVED** [review]

**Comment by SHeimlich** [comment]
\gcbrun

**Comment by SHeimlich** [comment]
\gcbrun

**Pull request closed without merging**